### PR TITLE
Restructure format

### PR DESCRIPTION
**Pull request by Simon-Laux**
doesn't parse the connection info yet (I'm working on it in my local working copy of this, but feel free to merge as I already started a separate branch for it)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,29 +7,16 @@ First check wether the file for the provider already exists, if it does skip to
 Add [provider name].md to the _providers folder in the following format:
 ~~~
 ---
+website: [website of the provider]
 name: [provider name]
-domains: [ email domains of the provider / can also be yaml array when there are multiple ones can contain (€) to mark a premium domain]
-credentials: [emailPass | emailAppPass | Oauth] - can also be an array, if multiple are supported availible
+domains: [email domains of the provider / must be an yaml array]
```
Comment (r10s) [inline README.md]: what is a yaml array? however, it is good to allow only one format

File: README.md [hunk 2]
```diff
@@ -7,29 +7,16 @@ First check wether the file for the provider already exists, if it does skip to
...
+before_login_hint: [~ or a string that should be displayed before the user logs in]
+after_login_hint: [~ or a string that should be displayed after the user logged in]
```
Comment (r10s) [inline README.md]: why not just leave this field out if it is not needed? i expect that to be true for the vast majority of providers.

File: README.md [hunk 3]
```diff
@@ -7,29 +7,16 @@ First check wether the file for the provider already exists, if it does skip to
 Add [provider name].md to the _providers folder in the following format:
 ~~~
 ---
+website: [website of the provider]
 name: [provider name]
```
Comment (r10s) [inline README.md]: nitpicking: i would leave name as the first item

File: README.md (hunk 1 above)
Comment (r10s) [reply README.md]: nm, found it. it is sth. as ['key1', 'key2', 'key3']

**Comment by r10s** [comment]
i've added an example to https://support.delta.chat/t/new-provider-db/732/6 - i think we should figure out the details first before we can really do some coding.

File: README.md (hunk 2 above)
Comment (pabzm) [reply README.md]: I would ask for the code to set this field to the absent value (e.g. `NULL`) in any of these cases:

* The field is not present,
* the value is `~`,
* the value is an empty string or a string that contains only whitespace.

Comment (Simon-Laux) [reply README.md]: I used serde yaml for parsing, it serialized this empty field to `~`, I didn't try leaving it out yet to see whether It'll be parsed as `None` or if it complains instead when the key is missing.

> the value is an empty string or a string that contains only whitespace.

that would require manual parsing, but we can add a check that prevents the build/ci from getting green in this case instead.

File: README.md (hunk 3 above)
Comment (Simon-Laux) [reply README.md]: order does not matter in yaml AFAIK.

Comment (pabzm) [reply README.md]: True, in YAML the order does not matter, but in the README we're writing for humans, not the YAML-parser. So it makes some sense to maintain an order of fields.

Comment (Simon-Laux) [reply README.md]: meh don't only look at the readme, it's the smallest part of the pr, also we can make it pretty later.

File: README.md (hunk 2 above)
Comment (Simon-Laux) [reply README.md]: Ok, just tested it, works also fine if the field is not present.

File: README.md (hunk 3 above)
Comment (pabzm) [reply README.md]: I'm waiting with reviewing the code until it's settled and cleanly committed. Until then I'm looking at text.

**Comment by pabzm** [comment]
Please rebase onto origin/simple-howto-style, that branch changed a little.

**Comment by Simon-Laux** [comment]
@pabzm I rebased it and gitignore the .bundle/config as well

**Pull request merged**